pkg/secrets/handlers/secrets_handler_k8s_test.go

```diff
@@ -29,4 +30,35 @@ func TestSecretsHandlerK8sUseCase(t *testing.T) {
 			So(err.Error(), ShouldEqual, "error map should not be empty")
 		})
 	})
+
+	Convey("updateK8sSecretsMapWithConjurSecrets", t, func () {
+		secret := []byte{'s', 'u', 'p', 'e', 'r'}
+		conjurSecrets := make(map[string][]byte)
+		conjurSecrets["account:variable:allowed/username"] = secret
+
+		newDataEntriesMap := make(map[string][]byte)
+		newDataEntriesMap["username"] = []byte("allowed/username")
+
+		k8sSecretsMap := make(map[string]map[string][]byte)
+		k8sSecretsMap["mysecret"] = newDataEntriesMap
+
+		pathMap := make(map[string]string)
+		pathMap["allowed/username"] = "mysecret:username"
+
+		k8sSecretsStruct := k8s.K8sSecretsMap{k8sSecretsMap, pathMap}
+		err := updateK8sSecretsMapWithConjurSecrets(&k8sSecretsStruct, conjurSecrets)
+
+		eq := reflect.DeepEqual(k8sSecretsStruct.K8sSecrets["mysecret"]["username"], secret)
+		Convey("Returns true if K8sSecretsMap's variable ID is replaced with corresponding secret value", func() {
+			So(err, ShouldEqual, nil)
+			So(eq, ShouldEqual, true)
+		})
+
+		Convey("Returns true if Conjur secret byte array is cleared from memory", func() {
```

Comment:
Regarding the 2 tests separation - i think this is quite good but I would add another Convey that wraps them. For example `Convey("One K8s secret with one Conjur secret"...` or something in that manner. It will be easier to add more scenarios to this test. please see the example here: http://goconvey.co/ (under Expressive DSL)

Reply:
Given When Then seems to be the structure they use. I could say:
`Given one K8s secret with one Conjur secret then the Variable ID for the Conjur secret is replaced in k8sSecretsMap with their corresponding secret value`

Reply:
I think we should go with `Given` -> `it` like we have in ruby UTs.

take this for an example:
```
Convey("updateK8sSecretsMapWithConjurSecrets", t, func () {
	Convey("Given one K8s secret with one Conjur secret", func() {
		// define all mocks

		err := updateK8sSecretsMapWithConjurSecrets(&k8sSecretsStruct, conjurSecrets)

		Convey("Finishes without raising an error", func() {
			So(err, ShouldEqual, nil)
		})

		Convey("Replaces the Variable ID with their corresponding secret value in k8sSecretsMap", func() {
			eq := reflect.DeepEqual(k8sSecretsStruct.K8sSecrets["mysecret"]["username"], secret)
			So(eq, ShouldEqual, true)
		})

		Convey("Clears the Conjur secret byte array from memory", func() {
			// secret value should be cleared from memory after use
			empty := make([]byte, len(secret))
			eq = reflect.DeepEqual(conjurSecrets["account:variable:allowed/username"], empty)
			So(eq, ShouldEqual, true)
		})
	})
})
```

Reply:
each thing we test has its own `Convey` section so we pin point test failures. this way once a test is failing we don't need to further investigate *what* failed in it. when you write the "it" convey imagine the sentence starts with "it"

Reply:
Is `Given` what is passed as a parm in the function? or just set the context?

Reply:
in general `given` is everything that happens before what you are actually testing. it is things you assume are working and you're not testing them now. they are given to you.